qutip/solver/solver_base.py

```diff
@@ -102,7 +102,10 @@ def _restore_state(self, data, *, copy=True):
             state = Qobj(data, **self._state_metadata, copy=copy)
 
         if data.shape[1] == 1 and self._options['normalize_output']:
-            state = state * (1 / state.norm())
+            if state._isherm:
```

Comment:
This is meant to check whether the state is a density matrix, yes? Surely there must be a better way if this doesn't always work. Also definitely worth a comment if the check remains a bit obscure.